platforms/software/dependency-management/src/main/java/org/gradle/api/internal/artifacts/type/DefaultArtifactTypeRegistry.java

```diff
@@ -58,6 +61,7 @@ public void visitArtifactTypes(Consumer<? super ImmutableAttributes> action) {
                 if (seen.add(artifactTypeDefinition.getName())) {
                     ImmutableAttributes attributes = ((AttributeContainerInternal) artifactTypeDefinition.getAttributes()).asImmutable();
                     attributes = attributesFactory.concat(attributesFactory.of(ARTIFACT_TYPE_ATTRIBUTE, artifactTypeDefinition.getName()), attributes);
+                    attributes = attributesFactory.concat(defaultArtifactAttributes.asImmutable(), attributes);
```

Comment:
:thinking: we're prepending the default attributes in many places. Maybe it is worth to try and abstract it somehow, so there's less risk of forgetting doing so on some codepath?

Reply:
Changed code a bit so we concat default attributes in fewer places